windows/userdata.ps1

```diff
@@ -226,25 +226,27 @@ Tfi-Out "Write userdata status file" $?
 netsh advfirewall firewall set rule name="WinRM in" new action=allow
 Tfi-Out "Open firewall" $?
 
-# if $Error variables has a queue of errors, this will output them to a file
-$ErrorLog = "C:\Temp\errors.log"
-Add-Content $ErrorLog -value "ERRORS --------------------"
-Add-Content $ErrorLog -value $Error | Format-List *
-
 # upload logs to S3 bucket
 $S3Keyfix="Win" + (((Get-WmiObject -class Win32_OperatingSystem).Caption) -replace '.+(\d\d)\s(.{2}).+','$1$2')
 If ($S3Keyfix.Substring($S3Keyfix.get_Length()-2) -eq 'Da') {
     $S3Keyfix=$S3Keyfix -replace ".{2}$"
 }
 
 $ArtifactPrefix = "${tfi_build_date}/${tfi_build_hour}_${tfi_build_id}/$S3Keyfix"
+Tfi-Out "Writing logs to $ArtifactPrefix"
+
+$ErrorActionPreference = "Continue"
 
-Write-S3Object -BucketName "${tfi_s3_bucket}/$ArtifactPrefix" -File ${tfi_win_userdata_log} -ErrorAction SilentlyContinue
-Write-S3Object -BucketName "${tfi_s3_bucket}/$ArtifactPrefix" -File $ErrorLog -ErrorAction SilentlyContinue
-Write-S3Object -BucketName "${tfi_s3_bucket}" -Folder "C:\\Program Files\\Amazon\\Ec2ConfigService\\Logs" -KeyPrefix $ArtifactPrefix/cloud/ -ErrorAction SilentlyContinue
-Write-S3Object -BucketName "${tfi_s3_bucket}" -Folder "C:\\ProgramData\\Amazon\\EC2-Windows\\Launch\\Log" -KeyPrefix $ArtifactPrefix/cloud/ -ErrorAction SilentlyContinue
-Write-S3Object -BucketName "${tfi_s3_bucket}" -Folder "C:\\Watchmaker\\Logs" -KeyPrefix $ArtifactPrefix/watchmaker/ -SearchPattern *.log -ErrorAction SilentlyContinue
-Write-S3Object -BucketName "${tfi_s3_bucket}" -Folder "C:\\Watchmaker\\SCAP\\Results" -KeyPrefix $ArtifactPrefix/scap_output/ -ErrorAction SilentlyContinue
+Test-Command "Write-S3Object -BucketName `"${tfi_s3_bucket}/$ArtifactPrefix`" -File `"${tfi_win_userdata_log}`""
+Test-Command "Write-S3Object -BucketName `"${tfi_s3_bucket}`" -Folder `"C:\\Watchmaker\\Logs`" -KeyPrefix `"$ArtifactPrefix/watchmaker/`" -SearchPattern `"*log`""
+Test-Command "Write-S3Object -BucketName `"${tfi_s3_bucket}`" -Folder `"C:\\Watchmaker\\SCAP\\Results`" -KeyPrefix `"$ArtifactPrefix/scap_output/results`" -Recurse"
```

Comment:
This changes the scap output location so it is different between windows and linux, scap_output vs scap_output/results. I'd like to keep them the same. Either scap/output on both (and scap/logs on windows), or scap_output and scap_logs.

Reply:
To avoid excess drill down, LX is now scap_output, win is scap_output/scap_logs.